dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AudioToText;
+using Microsoft.SemanticKernel.Contents;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel.Connectors.AssemblyAI;
+
+/// <summary>
+/// AssemblyAI speech-to-text service.
+/// </summary>
+public sealed class AssemblyAIAudioToTextService : IAudioToTextService
+{
+    private const string BaseUrl = "https://api.assemblyai.com/";
+    private readonly string _apiKey;
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Attributes is not used by AssemblyAIAudioToTextService.
+    /// </summary>
+    public IReadOnlyDictionary<string, object?> Attributes => new Dictionary<string, object?>();
+
+    /// <summary>
+    /// Creates an instance of the <see cref="AssemblyAIAudioToTextService"/> with an AssemblyAI API key.
+    /// </summary>
+    /// <param name="apiKey">OpenAI API Key</param>
+    /// <param name="httpClient"></param>
+    public AssemblyAIAudioToTextService(
+        string apiKey,
+        HttpClient? httpClient = null
+    )
+    {
+        this._apiKey = apiKey;
+        this._httpClient = HttpClientProvider.GetHttpClient(httpClient);
+    }
+
+    /// <inheritdoc />
+    public async Task<TextContent> GetTextContentAsync(
+        AudioContent content,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        string uploadUrl;
+        if (content.Uri is not null)
+        {
+            // to prevent unintentional file uploads by injection attack
+            if (content.Uri.IsFile)
+            {
+                throw new ArgumentException("File URI is not allowed. Use `AudioContent.Stream` or `AudioContent.File` to transcribe a local file instead.");
+            }
+
+            uploadUrl = content.Uri.ToString();
+        }
+        else if (content.Data is not null)
+        {
+            using var stream = content.Data!.ToStream();
+            uploadUrl = await this.UploadFileAsync(stream, cancellationToken).ConfigureAwait(false);
+        }
```

Comment:
It's a bit inconsistent with AudioContent where you gave priority to BinaryData and here you check the URI first.
`/// <summary>
    /// Returns the string representation of the audio.
    /// BinaryData audio will be represented as DataUri
    /// Remote Uri audio will be represented as is
    /// </summary>
    /// <remarks>
    /// When Data is provided it takes precedence over URI
    /// </remarks>
    public override string ToString()
    {
        return this.BuildDataUri() ?? this.Uri?.ToString() ?? string.Empty;
    }`